### PR TITLE
Designed post card

### PR DESCRIPTION
**Pull request by RossWebFS**
- Designed post card for different screens
- Added skeleton to post card component

### CODE REVIEW

File: src/pages/blog/BlogPage.tsx [hunk 1]
```diff
@@ -15,10 +16,11 @@ export default function BlogPage() {
       title="Blog"
       description="Blog with posts from experts in our field"
     >
-      <Box sx={{ marginTop: "100px" }}>
+      <Box sx={{ marginTop: "100px", backgroundColor: "#F8EBE1" }}>
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.tsx:19]: Let's use styled component here. Prefer using them where possible

File: src/pages/blog/components/post-card-skeleton/PostCardSkeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          <IconButton
+            disableRipple
+            sx={{
+              "&:hover": { backgroundColor: "transparent" },
+              "&:active": { backgroundColor: "transparent" },
+            }}
+            area-label="share"
+          >
+            <ShareIcon />
+          </IconButton>
```
Comment (mhevyk) [inline src/pages/blog/components/post-card-skeleton/PostCardSkeleton.tsx:42]: Good hob here! But I suggest removing one of skeleton content-like containers to make post with less height

image

File: src/pages/blog/components/post-card/PostCard.styled.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  minHeight: "170px",
+  backgroundColor: theme.palette.primary.main,
+  boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+  opacity: isLoading ? 0.3 : 1,
+  ...(shouldShowImagePlaceholder
+    ? { "& img": { width: "100px" } }
+    : { "& img": { width: "100%" } }),
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.styled.ts:48]:
```suggestion
"& img": {
    width: shouldShowImagePlaceholder ? "100px" : "100%"
}
```

File: src/pages/blog/components/post-card/PostCard.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
+import { Link } from "react-router-dom";
+
+import { Avatar, Box, IconButton, Typography } from "@mui/material";
+import { useMediaQuery } from "@mui/material";
+import { Icon } from "@mui/material";
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.tsx:5]: Prefer using only default imports from mui, it improves performance

https://mui.com/material-ui/guides/minimizing-bundle-size/

File: src/pages/blog/components/post-card/PostCard.tsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your audience to continue reading...",
+  viewsCount: 0,
+  commentsCount: 0,
+};
+// let post: Post = undefined;
+
+export default function PostCard() {
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down("sm"));
+
+  if (post === undefined) return <PostCardSkeleton />;
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.tsx:71]: Avoid using inline checks as it is one of code styles in the project. I think it should be added to eslint config as a rule

File: src/pages/blog/components/post-card/PostCard.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+  return (
+    <BoxStyled>
+      <Link to={`${routePaths.posts.path}/${post.id}`}>
+        <BoxImageStyled isLoading={false} shouldShowImagePlaceholder={hasError}>
+          <ImgStyled src={image} alt={`'${post?.title}' post image`} />
+        </BoxImageStyled>
+        <ContentBox>
+          <PostInfoBox>
+            <Box display="flex" gap={1}>
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.tsx:86]: Link has :active styles by default. Please fix strange behavior (I do right click):


https://github.com/user-attachments/assets/fbe6d373-1c40-4202-8f65-b95adc3d5737

File: src/pages/blog/components/post-card/PostCard.tsx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+              <Avatar />
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.tsx:87]: Avatar should redirect to member by id page

**Review by mhevyk — CHANGES_REQUESTED** [review]

File: src/pages/blog/BlogPage.tsx (hunk 1 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.tsx:19]: Fixed
Separated styles into BlogPage.styled.ts file
image

File: src/pages/blog/components/post-card-skeleton/PostCardSkeleton.tsx (hunk 2 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card-skeleton/PostCardSkeleton.tsx:42]: Fixed
Removed skeleton for title
image

File: src/pages/blog/components/post-card/PostCard.tsx (hunk 4 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.tsx:5]: Fixed imports from mui

image
image
image

File: src/pages/blog/components/post-card/PostCard.tsx (hunk 6 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.tsx:86]: Fixed

image

File: src/pages/blog/components/post-card/PostCard.tsx (hunk 7 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.tsx:87]: Fixed
Added link to author details

image

File: src/pages/blog/BlogPage.styled.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import Box from "@mui/material/Box";
+import { styled } from "@mui/material/styles";
```
Comment (mhevyk) [inline src/pages/blog/BlogPage.styled.ts:2]: import from `@mui/material`

File: src/pages/blog/components/post-card/PostCard.styled.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
+import Box from "@mui/material/Box";
+import IconButton from "@mui/material/IconButton";
+import { styled } from "@mui/material/styles";
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.styled.ts:3]: Same here, always use `@mui/material` for importing styled

File: src/pages/blog/components/post-card/PostCard.styled.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  minHeight: "170px",
+  backgroundColor: theme.palette.primary.main,
+  boxShadow: "0 4px 8px rgba(0, 0, 0, 0.1)",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+  opacity: isLoading ? 0.3 : 1,
+"& img": {
+    width: shouldShowImagePlaceholder ? "100px" : "100%"
+}
```
Comment (mhevyk) [inline src/pages/blog/components/post-card/PostCard.styled.ts:57]: Fix formatting (prettier)

**Comment by mhevyk** [comment]
image

Seems like responsiveness is broken. Can you please take a look?

File: src/pages/blog/BlogPage.styled.ts (hunk 8 above)
Comment (RossWebFS) [reply src/pages/blog/BlogPage.styled.ts:2]: Fixed

File: src/pages/blog/components/post-card/PostCard.styled.ts (hunk 9 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.styled.ts:3]: Fixed

File: src/pages/blog/components/post-card/PostCard.styled.ts (hunk 10 above)
Comment (RossWebFS) [reply src/pages/blog/components/post-card/PostCard.styled.ts:57]: Fixed

**Pull request merged**